### PR TITLE
Add Asset API docs

### PR DESCRIPTION
**Pull request by windy1**
This PR adds documentation for the new Asset API.

Signed-off-by: Walker Crouse <walkercrouse@hotmail.com>

### CODE REVIEW

**Comment by Spongy** [comment]
Setting up PR reference, please wait...

File: source/plugin/assets/index.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+The ``AssetManager`` allows developers to retrieve resources from from a plugin JAR.
+
+By default, a plugin's assets resides in a directory named ``assets/com/example/myplugin`` where
+``com.example.myplugin`` is the plugin ID. In most cases this default directory should be used but if you *need* to
+change it, you may do so by setting the ``assets`` attribute in your ``@Plugin`` annotation or ``mcmod.info`` file.
+
+Once properly configured you can retrieve a resource for your (or any) plugin using the following code:
+
+.. code:: java
```
Comment (Minecrell) [inline source/plugin/assets/index.rst]: `code-block`

File: source/plugin/assets/index.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+Working with Assets
+===================
+
+The ``Asset`` class is essentially just a wrapper around a ``URL`` with some common I/O operations built in. The use
+cases of ``Assets``\s is essentially unbounded but one common use case is to generate a default configuration file if
+your plugin's configuration file is not found. You can achieve this with the following code:
+
+.. code:: java
+
+    if (!Files.exists(configPath)) {
```
Comment (Minecrell) [inline source/plugin/assets/index.rst]: `Files.notExists`

File: source/plugin/assets/index.rst [hunk 3]
```diff
@@ -0,0 +1,50 @@
+=============
+The Asset API
+=============
+
+The ``AssetManager`` allows developers to retrieve resources from from a plugin JAR.
```
Comment (Inscrutable) [inline source/plugin/assets/index.rst]: Typo: "from from".

File: source/plugin/assets/index.rst [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    import org.spongepowered.api.asset.Asset;
+
+    Asset asset = plugin.getAsset("myfile.txt");
+
+Alternatively, you can retrieve assets through the ``AssetManager`` class:
+
+.. code-block:: java
+
+    import org.spongepowered.api.asset.AssetManager;
```
Comment (12AwsomeMan34) [inline source/plugin/assets/index.rst]: Should probably have the import for `Sponge`, which is used below.

File: source/plugin/assets/index.rst [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+Working with Assets
+===================
+
+The ``Asset`` class is essentially just a wrapper around a ``URL`` with some common I/O operations built in. The use
+cases of ``Assets``\s is essentially unbounded but one common use case is to generate a default configuration file if
+your plugin's configuration file is not found. You can achieve this with the following code:
+
+.. code-block:: java
+
```
Comment (12AwsomeMan34) [inline source/plugin/assets/index.rst]: Should also be an import for `Files`.

**Comment by windy1** [comment]
@Inscrutable @12AwsomeMan34 Fixed.

File: source/plugin/assets/index.rst [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+``com.example.myplugin`` is the plugin ID. In most cases this default directory should be used but if you *need* to
+change it, you may do so by setting the ``assets`` attribute in your ``@Plugin`` annotation or ``mcmod.info`` file.
+
+Once properly configured you can retrieve a resource for your (or any) plugin using the following code:
+
+.. code-block:: java
+
+    import org.spongepowered.api.asset.Asset;
+
+    Asset asset = plugin.getAsset("myfile.txt");
```
Comment (kashike) [inline source/plugin/assets/index.rst]: Should be noted that `plugin` is `PluginContainer`.

**Comment by Inscrutable** [comment]
This is just about ready @windy1 if you could just lend your opinion on the last outstanding quibble(s).

**Pull request merged**